src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -0,0 +1,762 @@
+        add_ro_properties(ov::device::properties.name(), supported_properties);
+        add_ro_properties(ov::device::priorities.name(), supported_properties);
+        return decltype(ov::supported_properties)::value_type(supported_properties);
+    } else if (EXEC_NETWORK_METRIC_KEY(SUPPORTED_METRICS) == name) {
+        auto metrics = default_ro_properties();
+        add_ro_properties(METRIC_KEY(SUPPORTED_METRICS), metrics);
+        add_ro_properties(METRIC_KEY(SUPPORTED_CONFIG_KEYS), metrics);
+        return to_string_vector(metrics);
+    } else if (EXEC_NETWORK_METRIC_KEY(SUPPORTED_CONFIG_KEYS) == name) {
+        return to_string_vector(m_cfg.GetSupported());
+    } else if (ov::device::properties == name) {
+        ov::AnyMap all_devices = {};
+        for (auto&& subnetwork : m_networks) {
+            ov::AnyMap device_properties = {};
+            if (all_devices.count(subnetwork._device) == 0) {
+                auto device_supported_metrics = subnetwork._network->get_property(METRIC_KEY(SUPPORTED_METRICS));
+                for (auto&& property_name : device_supported_metrics.as<std::vector<std::string>>()) {
+                    device_properties[property_name] = subnetwork._network->get_property(property_name);
+                }
+                auto device_supported_configs = subnetwork._network->get_property(METRIC_KEY(SUPPORTED_CONFIG_KEYS));
+                for (auto&& property_name : device_supported_configs.as<std::vector<std::string>>()) {
+                    device_properties[property_name] = subnetwork._network->get_property(property_name);
+                }
+                all_devices[subnetwork._device] = device_properties;
+            }
+        }
+        return all_devices;
+    } else if (ov::model_name == name) {
+        return decltype(ov::model_name)::value_type(m_name);
+    } else if (ov::loaded_from_cache == name) {
+        return decltype(ov::loaded_from_cache)::value_type{m_loaded_from_cache};
+    } else if (ov::optimal_number_of_infer_requests == name) {
+        unsigned int value = 0u;
+        for (auto&& desc : m_networks) {
+            value =
+                std::max(value,
+                         desc._network->get_property(ov::optimal_number_of_infer_requests.name()).as<unsigned int>());
+        }
+        return decltype(ov::optimal_number_of_infer_requests)::value_type{value};
+    } else if (ov::execution_devices == name) {
+        std::vector<std::string> device_names;
+        std::set<std::string> s;
+        for (auto&& subnetwork : m_networks) {
+            if (s.count(subnetwork._device) != 0)
+                continue;
+            s.insert(subnetwork._device);
+            device_names.push_back(subnetwork._device);
+        }
+        return decltype(ov::execution_devices)::value_type{device_names};
+    }
+    return m_cfg.Get(name);
+}
+
+const std::vector<ov::Output<const ov::Node>>& ov::hetero::CompiledModel::inputs() const {
+    return m_compiled_inputs;
```

Comment:
In case of API 1.0, `AddPreprocessing` transformation will be applied on device side and will change parameters of inputs/outputs (e.g. precision). To translate updated parameters on Hetero side, I need to replace original `m_inputs`/`m_outputs` with inputs/outputs from already compiled models

Reply:
> In case of API 1.0, `AddPreprocessing` transformation will be applied on device side and will change parameters of inputs/outputs (e.g. precision). To translate updated parameters on Hetero side, I need to replace original `m_inputs`/`m_outputs` with inputs/outputs from already compiled models

Inputs and outputs of original model should not be changed. They should the same in base class.

Reply:
For the moment it isn't possible because of API 1.0 `AddPreprocessing` that changes inputs/outputs attributes, so Hetero CompiledModel `inputs()`/`outputs()` doesn't translate actual information

So current idea is next - replace inputs/outputs from `ov::Model` with compiled submodels inputs/outputs. In case of API 2.0 there should be no changes by subplugins. 
Also it helps with import/export - instead of deserializing inputs/outputs it can be just restored from imported model, so no need to serialize by Hetero plugin

Reply:
https://github.com/openvinotoolkit/openvino/pull/18429/files#r1259443663